RecoParticleFlow/PFClusterProducer/interface/PFHGCalRecHitCreator.h

```diff
@@ -58,8 +59,15 @@ template <typename DET,PFLayer::Layer Layer,ForwardSubdetector subdet>
 	double energy = hgrh.energy();
 	double time = hgrh.time();	
 	
-	const FlatTrd *thisCell = 
-	  static_cast<const FlatTrd*>(hgcGeo.getGeometry(detid));	
+	const CaloCellGeometry *thisCell = nullptr;
+	
+	if( DetId::Forward == detid.det() ) {
+	  const auto* hgGeom = static_cast<const HGCalGeometry*>(geom->getSubdetectorGeometry(detid.det(),detid.subdetId()));
+	  thisCell = static_cast<const FlatTrd*>(hgGeom->getGeometry(detid));
```

Comment:
thisCell is of type CaloCellGeometry.
Why do you need a cast to something else?

Also, the getGeometry methods are apparently virtual. Is this if-else needed at all? It seems redundant on top of vtable walk that will happen anyways.